docs/content/sips/xxx-spin-plugins.md

```diff
@@ -0,0 +1,154 @@
+
+### The `install` subcommand
+
+When the `spin plugin install $plugin` command is invoked, spin uses a centralized repository to obtain information about the requested plugin.
+
+Once the information such as the source and version of the plugin is obtained, spin proceeds to download and place the bianry in a spin managed directory.  
+
+#### Update of plugins
+
+The system will use a rolling updates system where the latest plugin will be installed.
+
+### Centralized plugin Repository
+
+- A new github reposity `spin-plugins` will act as the index for all the plugins.
+- It will contain information about the plugins such as location of binary release for the various operating system, version and checksum along with potentially other details such as author details and a description.
+- Creators of new plugins can submit PRs with the required information to add it to the plugins index repository.
+
+The structure of data stored for every plugin is a json file with the following structure
+
+```
+{
+    "name": "test",
+    "description": "Some description.",
+    "homepage": "www.example.com",
+    "version": "1.0",
+    "license": "Mit",
+    "packages": [
+        {
+            "os": "linux",
+            "arch": "amd64",
+            "url": "<url_to_plugin_tar>",
+            "sha256": "18f023db8b39b567a6889fa6597550ab2b47f7947ca47792ee6f89dfc5a814e3"
+        },
+        {
+            "os": "macOs",
+            "arch": "aarch64",
+            "url": "<url_to_plugin_tar>",
+            "sha256": "18f023db8b39b567a6889fa6597550ab2b47f7947ca47792ee6f89dfc5a814e3"
+        }
+    ]
+}
+```
+
+The `packages` field is a list of all the package variants available for the plugin based on the different OS and architecture of the system. Not all plugins need to be able to run for all variants.
+
+The advantages of this method is that it will allow users to be able to search for plugins and imporve visibility of plugins. This will also allow for potentially implementation a `spin plugin search` subcommand in the future.
+
+#### General Naming Conventions of the plugins
+
+- The naming conventions for the binary in the releases must follow certain requirements:
+    - The release binary name should be `spin-$plugin`. For windows, it must contain the extension `.exe`.
+    - The release binary should not contain any version number in its name.
+
+#### Naming Conventions  of Spin Dependant plugins
+
+Spin dependant plugins can be distinguished from standalone binary plugins based on the name of the plugin. The Spin Dependant Plugins must follow the additional naming constraints.
+
+- The name of the plugin must be `spin-$plugin` where `$plugin` must be `<type_of_plugin>-<name_of_plugin>`
+
+Spin dependant plugins will be categorized based on the setup/functionality sharing required from the spin binary. Based on this, they will be grouped into `<type_of_plguin>`. Some of the possible types are 
+
+- Host components
+- Loaders
+- Config providers
```

Comment:
These examples are "internal plugins" right?  This is a super useful list - it would be good to have this up top where you define the three categories.

Reply:
Yes, these are examples of `spin-based internal plugins`.